### DISCUSSION TITLE
How to temporarily change a query param, and then change it back after successful fetch

### DISCUSSION THREAD

**Discussion by kmiwa007** [TanStack/query#6159] [Q&A]
Hi, I have a bit of an odd use case where I need to call an API with a param that changes under certain conditions. 
For ex.,
- First fetch needs `use_cache=false`
- All subsequent fetches need `use_cache=true`
- After a certain mutation, next fetch needs `use_cache=false`
- Then all subsequent fetches need `use_cache=true`

I tried doing this by making "use_cache" as React state that gets passes to useQuery() however I'm not sure how to set it back after the successful fetch.

**Comment by TkDodo** [comment]
you _can_ use the `meta` field with a `ref` and change it back and forth - `meta` gets passed into the queryFn:

```
const ref = useRef(false)

useQuery({
  queryKey,
  queryFn: async ({ meta }) => {
    const result = await fetchItems({ useCache: meta.ref.current })
    meta.ref.current = true
    return result
  },
  meta: {
    ref
  }
})
```

But really, you usually shouldn't need this. The client doesn't tell the backend to use a cache or not use a cache. This is a really suboptimal api design and imo the best course of action is to change that.
